src/renderer/components/ConfigureServer.tsx

```diff
@@ -64,28 +65,26 @@ function ConfigureServer({
     const [showContent, setShowContent] = useState(false);
     const [waiting, setWaiting] = useState(false);
 
-    const canSave = name && url && !nameError && !urlError;
+    const [validationResult, setValidationResult] = useState<URLValidationResult>();
+    const [validating, setValidating] = useState(false);
+
+    const canSave = name && url && !nameError && !urlError && !validating;
 
     useEffect(() => {
         setTransition('inFromRight');
         setShowContent(true);
     }, []);
 
-    const checkProtocolInURL = (checkURL: string): Promise<string> => {
-        if (isValidURL(checkURL)) {
-            return Promise.resolve(checkURL);
-        }
-        return window.desktop.modals.pingDomain(checkURL).
-            then((result: string) => {
-                const newURL = `${result}://${checkURL}`;
-                setUrl(newURL);
-                return newURL;
-            }).
-            catch(() => {
-                console.error(`Could not ping url: ${checkURL}`);
-                return checkURL;
+    useEffect(() => {
+        clearTimeout(validationTimeout);
```

Comment:
It is more idiomatic to clear in the effect return function instead of before doing anything in the effect. Also, it will make sure it gets cleared on unmount.